src/module.cpp

```diff
@@ -654,32 +632,66 @@ void module::print_cpp(std::ostream& os) const
 
 void module::annotate(std::ostream& os, std::function<void(instruction_ref)> a) const
 {
-    this->print([&](auto ins, const auto& names) {
-        print_instruction(os, ins, names);
+    this->print([&](auto ins, auto ins_names) {
+    	instruction::print(os, ins, ins_names);
         a(ins);
         os << std::endl;
     });
 }
 
+std::vector<module_ref> module::get_sub_modules() const
+{
+    std::vector<module_ref> vec_modules;
+    this->print([&](auto ins, auto) {
+        auto& mod_args = ins->module_inputs();
+        vec_modules.insert(vec_modules.end(), mod_args.begin(), mod_args.end());
+        for(auto& smod : mod_args)
+        {
+            auto sub_mods = smod->get_sub_modules();
+            vec_modules.insert(vec_modules.end(), sub_mods.begin(), sub_mods.end());
+        }
+    });
+
+    return vec_modules;
+}
+
 module& module::sort()
 {
     fix([&](auto self, auto ins) {
         this->move_instruction(ins, this->begin());
         for(auto child : ins->inputs())
+        {
             self(child);
+        }
     })(std::prev(this->end()));
     assert(this->validate() == this->end());
     return *this;
 }
 
+std::unordered_set<module_ref> module::get_parent_modules() const
+{
+    std::unordered_set<module_ref> parent_modules;
+    for(auto ins : iterator_for(*this))
+    {
+        auto& module_args = ins->module_inputs();
```

Comment:
This looks like the submodules not the parent modules.

Reply:
it was written when we have the parent_mod in the module class. looks like it is get_sub_modules now, so I just removed this function since we already have the get_sub_modules().